### ISSUE TITLE
"Generic" Andor cameras are detected before iXon 888

### ISSUE THREAD

**Issue by whooie** [AlexShkarin/pyLabLib-cam-control#2]
I wasn't able to work out why, but my camera was detected and loaded as "Generic Andor SDK2", and as a result I couldn't access any of the "Advanced" settings like trigger mode, shift speed, temperature control, etc.

I got around this by moving all of the definitions for `AndorSDK2IXONCameraDescriptor` (in `utils/cameras/AndorSDK2.py`) to the base class `AndorSDK2CameraDescriptor`, but other settings found in the Solis interface like "Acquisition Mode", "Readout Mode", and "Output Amplifier" (some others as well) are missing. Also, the options for "Horizontal shift speed", "Vertical shift period", and "Preamp gain" don't match the Solis interface.

**Comment by AlexShkarin** [comment]
Yes, I'm afraid that is currently the case. Andor SDK2 cameras have fairly different capabilites accessed not in a very uniform way. Hence, it's not trivial to make something universal, where I can be sure that it properly adapts to the camera capabilites but does not ask too much causing errors. That is why any Andor SDK2 camera other than Ixon and Luca falls back to "Generic Andor SDK2" camera with reduced capabilities access, and why you see different values for shift speeds and gains (these are hard-coded for Ixon rather than obtained from the camera).

I'm currently working on a more flexible approach, which makes better use of the data provided by the camera itself. Once I'm done, I can send you the updated code to try and see if it works any better.

**Comment by whooie** [comment]
I see, that makes sense. Happy to help test new features!
